www/md_base/guanlecoja/config.coffee

```diff
@@ -88,4 +88,33 @@ gulp.task 'icons', ->
         ))
         .pipe(gulp.dest(path.join(config.dir.build, 'icons')))
 
+gulp.task 'apiproxy', ->
+    # this is a task for developing, it proxy api request to http://nine.buildbot.net
+    argv = require('minimist')(process.argv)
+    argv.apiserver ?= 'nine.buildbot.net'
+    argv.apiport ?= 8021
+    argv.server ?= 'http://localhost:8020'
+
+    fs = require 'fs'
+    path = require 'path'
+    http = require 'http'
+    httpProxy = require 'http-proxy'
+    proxy = httpProxy.createProxyServer({})
+    proxy.on 'proxyReq', (proxyReq, req, res, options) ->
+        delete proxyReq.removeHeader('Origin')
+        delete proxyReq.removeHeader('Referer')
+    proxy.on 'proxyRes', (proxyRes, req, res) ->
+        proxyRes.headers['Access-Control-Allow-Origin'] = '*'
+        console.log "[Proxy] #{req.method} #{req.url}"
+
+    server = http.createServer (req, res) ->
+        if req.url.match /^\/api/
+            proxy.web req, res, {target: 'http://' + argv.apiserver}
+        else if req.url.match /^\/ws/
+            proxy.ws req, res, {target: 'ws://' + argv.apiserver}
+        else
+            proxy.web req, res, {target: argv.server}
```

Comment:
@tardyp 

I didn't quite get your idea. You mean using a script command to generate json of plugins and `configjson` and embed them into the page like what we do to `window.dataspec`?

It works if we are compiling the front end with the backend we are using, but it may not work when we'd like to precompile them into wheels. And how can we get the runtime `configjson`? Doesn't it need to modify the backend to provide an API entry?

BTW: how to specify `apiproxy` as a dev-only task in `config.coffee` to avoid the test failed due to this task will cause a timeout as the server will always running in the background.